[RA2 Ch03] - Tweak wording to cover both physical and virtual (#1894)
cntt-n/CNTT: doc/ref_arch/kubernetes/chapters/chapter03.md

[here](../../../ref_model/chapters/chapter05.md#54-nfvi-hw-profiles-features-and-requirements).

The Reference Model requires the support of Huge Pages in `i.cap.018` which is
already supported by upstream Kubernetes. For some applications, Huge Pages
should be allocated to account for consideration of the underlying HW topology.
This newer feature is missing from Kubernetes, therefore a gap has been
identified and added to [Chapter 6.2.8](./chapter06.md#628-hw-topology-aware-hugepages)
<a name="3.2.1.2"></a>
#### 3.2.1.2 HW Topology management

> This chapter should describe considerations about hardware topology
management.

<a name="3.2.1.3"></a>
#### 3.2.1.3 HW Acceleration

> This chapter should describe considerations about hardware acceleration, like
device management.

<a name="3.2.1.4"></a>
#### 3.2.1.4 CPU management

> This chapter should describe considerations about CPU management.

> Relate back to features described in the RM
[here](../../../ref_model/chapters/chapter05.md#521-virtual-compute) and
[here](../../../ref_model/chapters/chapter05.md#54-nfvi-hw-profiles-features-and-requirements).

<a name="3.2.1.5"></a>
#### 3.2.1.5 Container Runtime Services

> The Container Runtime is the component that runs within a Container Host

Reviewer: ```suggestion
The Container Runtime is the component that runs within a Kubernetes Node Operating System (OS) and manages the underlying OS functionality, such as cgroups and namespaces (in Linux), in order to provide a service within which container images can be executed and make use of the infrastructure resources (compute, storage, networking and other I/O devices) abstracted by the Kubernetes Node OS, based on API instructions from the kubelet.
```
I think we should consistently use Kubernetes Node.

Author: Fixed in https://github.com/cntt-n/CNTT/pull/1894/commits/710f0d7e5289c9a7559c2d6dabb4ecc80e46c76d (single line)

management of container images - moving them to where they need to be executed,
unpacking them, and then passing them to the low-level runtime, which then
executes the container. These high-level runtimes also include a comprehensive
API that other applications (e.g. Kubernetes) can use to interact and manage the
containers. An example of this type of runtime is containerd, which provides the
features described above, before passing off the unpacked container image to
runc for execution.

For Kubernetes the important interface to consider for container management is
the [Kubernetes Container Runtime Interface
(CRI)](https://kubernetes.io/blog/2016/12/container-runtime-interface-cri-in-kubernetes/).
This is an interface specification for any container runtime so that it is able
to integrate with the kubelet on a Kubernetes Node. The CRI decouples the
kubelet from the runtime that is running in the Host OS, meaning that the code
required to integrate kubelet with a container runtime is not part of the
kubelet itself (i.e. if a new container runtime is needed and it uses CRI, it
will work with kubelet). Examples of this type of runtime include containerd

> (with cri plugin) and cri-o, which is built specifically to work with

Reviewer 2: ```suggestion
(with CRI plugin) and cri-o, which is built specifically to work with
```

Kubernetes.

To fullfill `req.inf.vir.01` the architecture should support a container runtime
which provides the isolation of Operating System kernels.

The architecture must support a way to isolate the compute resources of the
infrastructure itself from the workloads compute resources.

<a name="3.2.2"></a>
### 3.2.2 Container Networking Services

As all production networking solutions for Kubernetes are based on CNI plugins
and the implicit requirement in [4.2.2 Virtual Network Interface
Specifications](../../../ref_model/chapters/chapter04.md#422-virtual-network-interface-specifications)
that documents the requirement to have the capability to attach several network
interfaces to the pods, the CNTT architecture must support a CNI metaplugin/CNI
multiplexer.

A CNI metaplugin/CNI multiplexer has the capability to attach several
interfaces, using different other CNI plugins, to a pod. Note that the different
network characteristics of the interfaces might require different networking
technologies, which would potentially require different CNI plugins.

> To comply with `req.inf.ntw.08`, inter node communication must be served by a
> CNI plugin which complies with the default K8s networking assumptions.

Reviewer 2: ```suggestion
To comply with `req.inf.ntw.08`, inter node communication must be served by a
CNI plugin which complies with the default [Kubernetes networking assumptions](https://kubernetes.io/docs/concepts/cluster-administration/networking/#the-kubernetes-network-model).
```
